src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NestedTypesShouldNotBeVisible.cs

```diff
@@ -81,6 +81,13 @@ public override void Initialize(AnalysisContext analysisContext)
                                 return;
                             }
 
+                            // Even if the nested type is declared public, don't complain if it's within
```

Comment:
I don't think so because a protected nested type would get SymbolVisibility.Public if I'm reading correctly and we explicitly want to allow nested protected types.